src/matches.rs

```diff
@@ -402,7 +402,16 @@ fn rewrite_match_body(
         let arrow_snippet = context.snippet(arrow_span).trim();
         // search for the arrow starting from the end of the snippet since there may be a match
         // expression within the guard
-        let arrow_index = arrow_snippet.rfind("=>").unwrap();
+        let mut arrow_index = arrow_snippet.rfind("=>").unwrap();
+        // check whether `=>` is included in the comment
+        if arrow_index != 0 {
+            let prev_arrow = arrow_snippet[..arrow_index].trim();
+            let single_line_comment_index = prev_arrow.rfind("//").unwrap_or(0);
+            let new_line_index = prev_arrow.rfind("\n").unwrap_or(0);
+            if single_line_comment_index > new_line_index {
+                arrow_index = 0;
+            }
+        }
```

Comment:
rustfmt has its own `FindUncommented` trait, which is implemented for `str`. Maybe you could consider simplifying this by using the `find_last_uncommented` method.

https://github.com/rust-lang/rustfmt/blob/73c81495cdb03d1fb73dd4313466c146ca8f4d98/src/comment.rs#L1122-L1163

Reply:
Fixed it, thanks!